connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SessionKey.java

```diff
@@ -23,14 +23,12 @@
 /**
  * A session key, which can be used to validate internal REST requests between workers.
  */
-public class SessionKey {
-
-    private final SecretKey key;
-    private final long creationTimestamp;
+public record SessionKey(SecretKey key, long creationTimestamp) {
```

Comment:
Similar to https://github.com/apache/kafka/pull/19452/files#r2041156088
We could use the compact constructor to minimize the code.

We could transfer the javadocs of getters to the class-level `@param` docs and then remove the redundant getter.